src/file/FileUtils.js

```diff
@@ -434,19 +434,24 @@ define(function (require, exports, module) {
      * @return {number} The result of the local compare function
      */
     function compareFilenames(filename1, filename2, extFirst) {
-        var ext1   = getFileExtension(filename1),
-            ext2   = getFileExtension(filename2),
-            lang   = brackets.getLocale(),
-            cmpExt = ext1.toLocaleLowerCase().localeCompare(ext2.toLocaleLowerCase(), lang, {numeric: true}),
-            cmpNames;
+        var lang = brackets.getLocale();
         
-        if (brackets.platform === "win") {
-            filename1 = getFilenameWithoutExtension(filename1);
-            filename2 = getFilenameWithoutExtension(filename2);
+        function cmpExt() {
+            var ext1 = getFileExtension(filename1),
+                ext2 = getFileExtension(filename2);
+            return ext1.toLocaleLowerCase().localeCompare(ext2.toLocaleLowerCase(), lang, {numeric: true});
+        }
+        
+        function cmpNames() {
+            if (brackets.platform === "win") {
```

Comment:
Memoizing helps when a function is called many times with same parameter vales. During a call to `.sort()` the same 2 *fullpaths* should never be compared more than once. This function compares just the name partion of the fullpath, so common names such as **index.html** will be compared more than once, but I think this is an exception, So the initial call will have the overhead of memoizing but very little of the benefits. It should help on subsequent calls to `.sort()`. But, when sorting all of the file in a file system, the combination of parameter values is huge, so I am not sure memoization is worth the overhead. Once again, it needs to be profiled.